hilo.rb

```diff
@@ -1,34 +1,111 @@
 #!/usr/bin/env ruby
-puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
-puts "You will have a total of 6 guesses."
-puts ""
 
-random_num = rand(1..100)
+# Game to guess random number
+class HiLoGame
 
-# gives the user 6 guesses
-for i in 0..5
-    guesses_left = 6 - i
-    puts "You have #{guesses_left} guesses left."
-    puts "Please guess a whole number."
+    # Constructor to initialise the class with a maximum number of guesses
+    def initialize(max_guesses)
+	@max_guesses = max_guesses
+	@already_guessed = Array.new
+	@random_num
+	@has_won = false
+    end
+
+    # The game is started
+    def play
+	welcome
+	@random_num = get_random_num
+
+	# Gives the user max number of guesses
+	for i in 0...@max_guesses
+	    if @has_won
+		print_has_won
+		return
+	    end
```

Comment:
You've still got a bit of duplication here, as you're doing the same thing down below to. I think using a `while` loop will be much neater than a `for` loop. Think something like
```
while game_running?
....
end
```